sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -979,22 +979,37 @@ private static void throwForTypeMismatches(List<TypeMismatch> mismatches) {
   private static List<PropertyDescriptor> validateClass(Class<? extends PipelineOptions> iface,
       Set<Class<? extends PipelineOptions>> validatedPipelineOptionsInterfaces,
       Class<? extends PipelineOptions> klass) throws IntrospectionException {
-    Set<Method> methods = Sets.newHashSet(IGNORED_METHODS);
-    // Ignore synthetic methods
-    for (Method method : klass.getMethods()) {
-      if (Modifier.isStatic(method.getModifiers()) || method.isSynthetic()) {
-        methods.add(method);
-      }
-    }
-    // Ignore methods on the base PipelineOptions interface.
-    try {
-      methods.add(iface.getMethod("as", Class.class));
-      methods.add(iface.getMethod("populateDisplayData", DisplayData.Builder.class));
-    } catch (NoSuchMethodException | SecurityException e) {
-      throw new RuntimeException(e);
-    }
-
     // Verify that there are no methods with the same name with two different return types.
+    validateReturnType(iface);
+
+    SortedSet<Method> allInterfaceMethods = FluentIterable
+        .from(ReflectHelpers.getClosureOfMethodsOnInterfaces(
+            validatedPipelineOptionsInterfaces))
+        .append(ReflectHelpers.getClosureOfMethodsOnInterface(iface))
+        .filter(NOT_SYNTHETIC_PREDICATE)
+        .toSortedSet(MethodComparator.INSTANCE);
+
+    List<PropertyDescriptor> descriptors = getPropertyDescriptors(allInterfaceMethods, iface);
+
+    // Verify that all method annotations are valid.
+    validateMethodAnnotations(allInterfaceMethods, descriptors);
+
+    // Verify that each property has a matching read and write method.
+    validateGettersSetters(iface, descriptors);
+
+    // Verify all methods are known.
+    validateMethodsAreKnown(iface, klass, descriptors);
+
+    return descriptors;
+  }
+
+  /**
+   * Validates that any property with the same name must have the same return type for all derived
```

Comment:
This is for method names and not properties since getters/setters for the same property would differ since setter would be void and getter wouldn't be void.